core/src/main/scala/kafka/log/Log.scala

```diff
@@ -792,6 +798,34 @@ class Log(@volatile var dir: File,
     deleteOldSegments(shouldDelete)
   }
 
+  private def updateAppendInfoForDuplicateEntry(appendInfo: LogAppendInfo) : LogAppendInfo = {
+    val segmentIterator = segments.descendingMap().values().iterator()
```

Comment:
nit: usually we leave off the parenthesis if the method doesn't mutate state. So `segments.descendingMap.values.iterator` works also.